src/_pytest/runner.py

```diff
@@ -258,7 +258,7 @@ def from_call(cls, func, when, reraise=None) -> "CallInfo":
         precise_start = perf_counter()
         try:
             result = func()
-        except:  # noqa
+        except BaseException:
```

Comment:
if we catch it here, we can as a followup clean up the exceptioninfo usage, and bind a variable name + use isintstance

Reply:
> clean up the exceptioninfo usage

It would be be nice to have a `ExceptionInfo.from_exception(e)` which does `ExceptionInfo.from_exc_info((type(e), e, e.__traceback__))`, but since it doesn't exist yet the current code seems a bit cleaner.

Reply:
all we use it for is a reraise, thats why i think we do not need the actual object at all
```
except ... as e:
  if reraise is not None and isinstance(e, reraise):
    raise
```

Reply:
oh eh im wrong, its necessary just below

however on pytohn3 we should deprecate from_current, and add a from_exception method, as that old style 3-tuple is no longer sensible and necessary

definitively something for a followup

i want to see a complete replacement for exceptioninfo mit to long term